lib/SPIRV/libSPIRV/spirv.hpp

```diff
@@ -71,21 +71,18 @@ enum ExecutionModel {
     ExecutionModelFragment = 4,
     ExecutionModelGLCompute = 5,
     ExecutionModelKernel = 6,
-    ExecutionModelCount, /* internal use only */
```

Comment:
It is only used in SPIRVEnum.h:

    inline bool
    isValid(SPIRVExecutionModelKind E) {
        return (unsigned)E < (unsigned)ExecutionModelCount;
    }

Should I do something like:

    inline bool
    isValid(SPIRVExecutionModelKind E) {
        return (unsigned)E <= (unsigned)ExecutionModelKernel;
    }

But this is quite messy as, if another `ExecutionModel` is later added, it will be really difficult to find all the above hacks in the code to update them. A (better) solution would be to redefine `ExecutionModelCount` locally to `ExecutionModelKernel + 1`, but this is hacky as well. What would your advice be?